aplib.net-demo/Assets/Scripts/Entities/Enemy/RangedEnemy.cs

```diff
@@ -44,6 +46,8 @@ protected override void Start()
     /// </summary>
     protected override void Update()
     {
+        Debug.DrawRay(transform.position, Vector3.up * 20, Color.green);
+
         // If the target is not within vision range, do nothing.
         if (!_pathFind.GoalWithinRange(_visionRange))
         {
```

Comment:
hele update kan anders met nieuwe pathfind, movecloser coroutine kan weg
```suggestion
    protected override void Update()
    {
        Debug.DrawRay(transform.position, Vector3.up * 20, Color.green);

        // If the target is not directly visible but within vision range, move closer to the target.
        if (!_rangedWeapon.EnemiesInLineOfSight() && _pathFind.GoalWithinRange(_defaultVisionRange))
        {
            _pathFind.SetStoppingDistance(1f);
        }

        if (_attackTimer.IsFinished() && _rangedWeapon.EnemiesInLineOfSight())
        {
            _pathFind.SetStoppingDistance(_attackRange - 1f);
            _rangedWeapon.UseWeapon();
            _attackTimer.Reset();
        }

        base.Update();
    }
```